scipy/stats/_stats_py.py

```diff
@@ -9244,41 +9244,47 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
     .. [8] https://en.wikipedia.org/wiki/Extensions_of_Fisher%27s_method
 
     """
-    if pvalues.size == 0:
+    xp = array_namespace(pvalues)
+    pvalues = xp.asarray(pvalues)
+    if xp_size(pvalues) == 0:
         NaN = _get_nan(pvalues)
         return SignificanceResult(NaN, NaN)
 
     if method == 'fisher':
```

Comment:
This is all pretty standard, so not much to say, but
```suggestion
    n = pvalues.shape[0]
    one = xp.asarray(1, dtype=pvalue.dtype)
    
    if method == 'fisher':
```
would help readability below (e.g. `2 * n * one` instead of `xp.asarray(2 * xp_size(pvalues), dtype=pvalues.dtype)`. Another possibility would be to let `_SimpleChi2` accept Python scalars as shapes and convert them to the dtype of the statistic argument as needed before evaluating the special function.

Also, as a nit, I think conceptually `.shape[0]` is more correct than `xp_size` because if this were vectorized, it would need to be `.shape[axis]`